lib/src/util/barback.dart

```diff
@@ -34,20 +34,38 @@ class BuildStepTransform implements barback.Transform {
   barback.Asset get primaryInput => toBarbackAsset(buildStep.input);
 
   @override
-  barback.TransformLogger get logger =>
-      new barback.TransformLogger((asset, level, message, span) {
-        var buffer = new StringBuffer();
-        buffer.write('$level: ');
-        if (asset != null) {
-          buffer.write('From $asset');
-          if (span != null) {
-            buffer.write('()$span)');
-          }
-          buffer.write(': ');
+  barback.TransformLogger get logger {
+    _logger ??= new barback.TransformLogger((asset, level, message, span) {
+      var buffer = new StringBuffer();
+      if (asset != null) {
+        buffer.write('From $asset');
+        if (span != null) {
+          buffer.write('()$span)');
         }
-        buffer.write(message);
-        print(buffer);
-      });
+        buffer.write(': ');
+      }
+      buffer.write(message);
+      switch (level) {
```

Comment:
Personal preference, but I'd rather see a map from `barback.LogLevel` to `Level` and then call `Logger#log`